### ISSUE TITLE
403 (Forbidden) -- JWT is not configurated properly, please contact the admin

### ISSUE THREAD

**Issue by johandelforge** [Tmeister/wp-api-jwt-auth#59]
Hi !

Getting currentUser is working fine, and validating the token also.
Htaccess has been modified & config.headers updated with Bearer xxx, otherwise validating the token would not work.  

But all my $http GET & POST are now giving me that 403 forbidden response... WP logged in or not
So now I'm adding the config.headers with Bearer only for POSTs, so GETs are back to working.  

Without the plugin, GETs, & POSTs when WP logged in were working great.  When not WP logged in, POSTs were not working (401 unauthorized)

... I thought i could make POSTs working when not WP logged in with the plug in...   with no luck.

Any tips are welcome...
Cheerz...

Johan

**Comment by kay14** [comment]
Having the same issue here on my side. This is the error that i am getting when trying to make a GET request with the jwt token included in the Header section.

{
    "code": "jwt_auth_bad_config",
    "message": "JWT is not configurated properly, please contact the admin",
    "data": {
        "status": 403
    }
}

Any suggestion as to what is going wrong here? Need help ASAP.
Thanks in advance

**Comment by Orabig** [comment]
TL;DR;
Move the definition of `JWT_AUTH_SECRET_KEY` to the top of your `wp-config.php` file

Detailed solution :

As stated in the installation doc of the plugin, 
the JWT needs a secret key to sign the token this secret key must be unique and never revealed.

Thus, you must add this line into your wp-config.php file

    define('JWT_AUTH_SECRET_KEY', 'your-top-secrect-key');

However, what the doc does not say, is that **you should not put this at the end of the file**. 
Insert this line just after the other definitions (AUTH_KEY, SECURE_AUTH_KEY, ... NONCE_SALT), and the 403 error should be gone.

I guess that's because the definition of `JWT_AUTH_SECRET_KEY` must be done before require_once(ABSPATH . 'wp-settings.php');

**Comment by RicardoRamirezR** [comment]
I got the same issue, and it was resolved changing in `wp-content/plugins/jwt-authentication-for-wp-rest-api/public/class-jwt-auth-public.php`:
```php
if (!$secret_key) {
```
by
```php
if ($secret_key === false) {
```
Put `define('JWT_AUTH_SECRET_KEY', 'your-top-secrect-key');` wherever you what but before `require_once(ABSPATH . 'wp-settings.php');` as @Orabig says.

**Comment by hellocaio** [comment]
@RicardoRamirezR, that worked for me, thanks! Weird tho...

**Comment by kuworking** [comment]
Worked for me, locating JWT_AUTH_SECRET_KEY as @Orabig says, no need (so far) to change the `if...` statement

**Comment by BehinaGroup** [comment]
Worked for me to, locating JWT_AUTH_SECRET_KEY before `require_once(ABSPATH . 'wp-settings.php');` as @Orabig says, no need (so far) to change the `if...` statement

**Comment by vainolo** [comment]
Had this same problem! Thanks for posting the issue and the solution

**Comment by smahi** [comment]
@Orabig 
Worked for me, thx

**Comment by crifan** [comment]
as @Orabig says, in `wp-config.php`, put:
```php
define('JWT_AUTH_SECRET_KEY', 'xxx');
```

before

```php
require_once(ABSPATH . 'wp-settings.php');
```

like this:

![2DED2C86-A506-4384-BF61-2E2D090FFFEB](https://user-images.githubusercontent.com/2750682/76142211-6a270080-60a6-11ea-91af-b8dac1c81e97.png)

can fix this issue.

**Comment by Osamasomy** [comment]
**sorry, You are no allowed to create posts as this user**

I always see this message when ever i try to create new post.
Im trying above solutions but still not getting success. :-(

Im also want to add some more fields while creating the product is this possible or can i create custom API for this?

**Comment by ngallazzi** [comment]
Got same issue, @Orabig solution works as a charm! Thanks!

**Comment by melomontoya** [comment]
This might help if you guys are on AWS Lightsail, it saved my day might save yours
https://www.youtube.com/watch?v=pmfVL3yeq84

**Comment by LouisAldorio** [comment]
How to fix this if I use wordpress docker image ?, I have to rebuild the image ?
